### DISCUSSION TITLE
Decentralizing the Installation Media Download

### DISCUSSION THREAD

**Discussion by leoherzog** [elementary/website#3379] [Ideas]
Linux distros commonly employ a network of HTTP mirrors for their media downloads, often volunteer education and nonprofits who donate bandwidth. I know elementary uses [Amazon S3 buckets](https://github.com/elementary/website/blob/5.0/_backend/classify.functions.php#L3-L10) for distribution, but is there any interest in using a web downloader powered by [WebTorrent](https://webtorrent.io/) on the homepage? If we were to implement that, Elementary could host one or two Webseed servers in a similar style to the current system, but in addition, any user that's downloading the ISO on the homepage will also help seed the file to others, right in the browser. This works across domains via WebRTC and is a really neat technical achievement.

**Comment by lewisgoddard** [comment]
We have looked at [implementing webtorrent before](https://github.com/elementary/website/pull/1275), but have always ended up stuck against things like [blob sizes not handling multi-GB](https://github.com/webtorrent/webtorrent/issues/319) or [service workers restarting after 10m](https://github.com/jimmywarting/StreamSaver.js/issues/39). I don't think we'd be opposed to moving from the current infrastructure (some Digital Ocean droplets with Nginx and Transmission) to webtorrent since we were looking before, but it needs to be a lot more reliable than we last found it in 2019.

**Reply by leoherzog** [reply]
Very cool, I hadn't come across PR #1275 while searching. Thanks for your thoughts.

I'm happy to report that last year, WebTorrent switched to using a hybrid chunk store in the browser instead of memory, which should help with your first issue.

https://github.com/webtorrent/webtorrent/pull/2208

I would assume that this helps the second issue as well, but I can't confirm.

We could also probably switch to using something like libtorrent on the DigitalOcean seedboxes. They now have WebTorrent support in their master branch with the `-DWebtorrent=on` build flag.

https://feross.org/libtorrent-webtorrent/

https://github.com/arvidn/libtorrent/pull/4123
